common/dynamicconfig/constants.go

```diff
@@ -2077,6 +2077,13 @@ const (
 	// Allowed filters: DomainName
 	EnableStrongIdempotencySanityCheck
 
+	// MatchingEnableAutoConfig enables auto config for domain and tasklist on worker side
+	// KeyName: matching.enableAutoConfig
+	// Value type: Bool
+	// Default value: false
+	// Allowed filters: DomainName,TasklistName,TasklistType
+	MatchingEnableAutoConfig
```

Comment:
nit: `ClientEnableAutoConfig` to avoid confusion with matching and also distinguish it from `cadence-worker` service.

Reply:
How about MatchingEnableClientAutoConfig, since this config is bind with the task list manager.

Reply:
make sense